src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -444,11 +446,29 @@ protected function getEnv($name)
         if (false !== $env = getenv($name)) {
             return $this->envCache[$name] = $env;
         }
-        if (!$this->hasParameter("env($name)")) {
+        if (!$this->isCompiled() && $this->parameterBag instanceof EnvPlaceholderParameterBag) {
```

Comment:
I don't think that's correct: when the container is not compiled, env values should not be used - only placeholders should be, so that we can latter on know where an env has been injected.